java/arcs/core/tools/Proto2Schema.kt

```diff
@@ -0,0 +1,138 @@
+package arcs.core.tools
+
+import arcs.core.data.FieldType
+import arcs.core.data.Manifest
+import arcs.core.data.Schema
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.arguments.multiple
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.file
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.PropertySpec
+import java.io.File
+
+class Proto2Schema : CliktCommand(
+    help = """Generates Schemas and Types from a protobuf-serialized manifest.
+    
+    This script reads schemas from a serialized manifest and generates Kotlin `Schema` and `Type` 
+    classes.""",
+    printHelpOnEmptyArgs = true
+) {
+
+    val outdir by option(help = "output directory; defaults to '.'").file(fileOkay = false)
+    val outfile by option(help = "output filename; if omitted")
+    val packageName by option(help = "scope to specified package; default: 'arcs'").default("arcs")
+    val protos by argument(help = "paths to protobuf-serialized manifests")
+        .file(exists = true).multiple()
+
+    /** Mock Proto2Schema: So far, this is equivalent to copy. */
+    override fun run() {
+        protos.forEach { protoFile ->
+            val outFile = outputFile(protoFile)
+            echo("$protoFile --> $outFile")
+
+            val bytes = protoFile.readBytes()
+            val manifest = Manifest.parseFrom(bytes)
+
+            outFile.writeBytes(protoFile.readBytes())
+        }
+    }
+
+    /** Produces a File object per user specification, or with default values. */
+    fun outputFile(inputFile: File): File {
+        val outputName = outfile ?: inputFile.nameWithoutExtension + ".kt"
+        val outputPath = outdir ?: System.getProperty("user.dir")
+        return File("$outputPath/$outputName")
+    }
+
+    fun formatName(name: String): String {
+        return name[0].toLowerCase() + name.substring(1)
+    }
+
+    fun generateSchemas(schemas: List<Schema>): Iterable<PropertySpec> {
```

Comment:
I don't know why you generate schemas from the objects that are already schemas. It would seem more productive to generate them from the proto. WDYT?

Reply:
The schema protos didn't exist at the time that I wrote this PR (I will sync now). I wanted to get started on learning how to do generation in KtPoet asap. 
I'll revise this to use the protos that you landed.